### PR TITLE
Add page migration tests with normal pages,hugepages and thp

### PR DESCRIPTION
**Pull request by harish-24**
This test validates migration of
a) normal pagesize
b) hugepages
c) hugepages with overcommit
d) transparent hugepages

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
``` (1/4) memory/migrate_pages.py:MigratePages.test;normal_page-5079: PASS (11.97 s)
 (2/4) memory/migrate_pages.py:MigratePages.test;hugepage-715d: PASS (2.39 s)
 (3/4) memory/migrate_pages.py:MigratePages.test;overcommit_huge-b107: PASS (2.06 s)
 (4/4) memory/migrate_pages.py:MigratePages.test;thp-8706: PASS (18.97 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

File: memory/migrate_pages.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    4) transparent hugepages
+    """
+
+    def copyutil(self, file_name):
+        shutil.copyfile(self.get_data(file_name),
+                        os.path.join(self.teststmpdir, file_name))
+
+    def setUp(self):
+        smm = SoftwareManager()
+        dist = distro.detect()
```
Comment (narasimhan-v) [inline memory/migrate_pages.py]: they can be called just before they are used

File: memory/migrate_pages.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        for file_name in ['node_move_pages.c', 'Makefile']:
+            self.copyutil(file_name)
+
+        build.make(self.teststmpdir)
+
+    def test(self):
+        os.chdir(self.teststmpdir)
+        cmd = './node_move_pages -n %s' % self.nr_chunks
+        if self.hpage:
+            cmd = '%s -h' % cmd
```
Comment (narasimhan-v) [inline memory/migrate_pages.py]: cmd += " -h"

File: memory/migrate_pages.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if self.hpage_commit:
```
Comment (narasimhan-v) [inline memory/migrate_pages.py]: shouldn't this be `if .. elif` blocks ? Will having both hpage and hpage_commit not cause problems ?

Comment (harish-24) [reply memory/migrate_pages.py]: Acutally -o is applicable only if hugepages (-h) are provided. with the above change `cmd += ' -h'` i will move the `hpage_commit` if block inside the `hpage if` block and add elif for `thp` block.

**Comment by harish-24** [comment]
@narasimhan-v Pushed with requested changes. Thank you for the review.
```
2019-05-21 10:48:23,640 test             L0602 INFO | START 2-memory/migrate_pages.py:MigratePages.test;hugepage-715d
...
2019-05-21 10:48:24,527 process          L0626 INFO | Running './node_move_pages -n 2 -h'
2019-05-21 10:48:25,152 process          L0458 DEBUG| [stdout] Total memory size 292756586496 bytes
2019-05-21 10:48:25,152 process          L0714 INFO | Command './node_move_pages -n 2 -h' finished with 0 after 0.622246980667s
...

2019-05-21 10:48:25,323 test             L0602 INFO | START 3-memory/migrate_pages.py:MigratePages.test;overcommit_huge-33d9
...
2019-05-21 10:48:26,278 process          L0626 INFO | Running './node_move_pages -n 2 -h -o'
2019-05-21 10:48:26,889 process          L0458 DEBUG| [stdout] Total memory size 292739088384 bytes
2019-05-21 10:48:26,889 process          L0714 INFO | Command './node_move_pages -n 2 -h -o' finished with 0 after 0.608815908432s
```

**Pull request merged**